core/dbt/task/compile.py

```diff
@@ -56,12 +57,26 @@ def runtime_cleanup(self, selected_uids):
         self._skipped_children = {}
         self._raise_next_tick = None
 
-    def handle_request(self, name, sql, macros=None):
+    def _extract_request_data(self, data):
+        data = self.decode_sql(data)
+        macro_blocks = []
+        data_chunks = []
+        for block in extract_toplevel_blocks(data):
+            if block.block_type_name == 'macro':
+                macro_blocks.append(block.full_block)
+            else:
+                data_chunks.append(block.full_block)
```

Comment:
what does the faux "block" look like here?

Reply:
Those blocks are defined here: https://github.com/fishtown-analytics/dbt/blob/0a9ed9977/core/dbt/clients/_jinja_blocks.py#L10-L15

 If it's a "real block", like `{% if ... %} ... {% endif %}` it is the exact contents of the block from open to close. Raw/comment blocks are kind of special, but not really special, and end up as 'data blocks'.

However, your comment just made me realize that the current code loses top-level comments and `raw` blocks. That seems pretty bad! I've added a fix for that and a corresponding test.